### PR TITLE
Introduce SAMLSSOPersistenceManagerFactory to switch between different persistence implementation

### PR DESCRIPTION
**Pull request by Osara-B**
### Proposed changes in this pull request

$subject. Related to,
* https://github.com/wso2/product-is/issues/20992

### CODE REVIEW

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/SAMLSSOServiceProviderManager.java [hunk 1]
```diff
@@ -29,7 +29,8 @@
  */
 public class SAMLSSOServiceProviderManager {
```
Comment (darshanasbg) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/SAMLSSOServiceProviderManager.java:30]: Lets rename to `SAMLServiceProviderPersistenceManager`

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactory.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+
+/**
+ * Factory class to create instances of SAMLSSOServiceProviderDAO based on the configured storage type.
+ */
+public class SAMLSSOPersistenceManagerFactory {
```
Comment (darshanasbg) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactory.java:29]: Lets rename to `SAMLServiceProviderPersistenceManagerFactory`

Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactory.java]: Resolved in [Introduce SAMLSSOServiceProviderDAO Factory class](https://github.com/wso2/carbon-identity-framework/pull/6240/commits/2c347ada3cf2d0754035586e395ccb929e51f1f4)

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactory.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+        SAMLSSOServiceProviderDAO samlSSOServiceProviderDAO = new RegistrySAMLSSOServiceProviderDAOImpl();
+        if (StringUtils.isNotBlank(SAML_STORAGE_TYPE)) {
+            switch (SAML_STORAGE_TYPE) {
+                case HYBRID:
+                    //Initialize hybrid SAML storage
+                    LOG.info("Hybrid SAML storage initialized.");
+                    break;
+                case DATABASE:
+                    //Initialize JDBC SAML storage
```
[the remark this answers is not in the document]
Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactory.java]: [Introduce SAMLSSOServiceProviderDAO Factory class](https://github.com/wso2/carbon-identity-framework/pull/6240/commits/2c347ada3cf2d0754035586e395ccb929e51f1f4)

File: components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactoryTest.java [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    }
+
+    private void setPrivateStaticField(Class<?> clazz, String fieldName, Object newValue)
+            throws NoSuchFieldException, IllegalAccessException {
+
+        Field field = clazz.getDeclaredField(fieldName);
+        field.setAccessible(true);
+        field.set(null, newValue);
+    }
+}
```
[the remark this answers is not in the document]
Comment (Osara-B) [reply components/identity-core/org.wso2.carbon.identity.core/src/test/java/org/wso2/carbon/identity/core/dao/SAMLSSOPersistenceManagerFactoryTest.java]: [Introduce SAMLSSOServiceProviderDAO Factory class](https://github.com/wso2/carbon-identity-framework/pull/6240/commits/2c347ada3cf2d0754035586e395ccb929e51f1f4)

**Review by UdeshAthukorala — APPROVED** [review]

**Review by darshanasbg — APPROVED** [review]

**Pull request merged**